client/blocks/inline-help/popover.jsx

```diff
@@ -422,7 +437,7 @@ function mapStateToProps( state ) {
 		selectedEditor: getSelectedEditor( state, siteId ),
 		classicUrl: `/${ classicRoute }`,
 		siteId,
-		showOptOut: optInEnabled && isGutenbergEditor,
+		showOptOut: optInEnabled && isGutenbergEditor && ! isUsingGutenbergPageTemplates,
```

Comment:
did we walk back from using the opt-in API to determine opt-in eligibility? I know at one point in time the design was supposed to be, "if there is an `opt-in` property in the API response then it means you can opt-in"

Reply:
do you mean the `/gutenberg` endpoint? I don't think that this logic can be translated to changes on that endpoint without making it a bit weird, since it just returns the editor for the platform, and it really just governs opt-in, not opt-out.
We can't lie and just return `classic` because it could have unintended consequences, and I don't think it's a good idea to add another field to the response and change the spirit of the endpoint.

Am I missing something? I personally don't like this solution and I'm open to anything, but it's really hard to tell what a _new_ site is.

Reply:
I lost familiarity with the code but when it was designed it was intended that the API response convey the editor selection and also whether or not someone was eligible for opt-in and opt-out

p7jreA-1Qf-p2

We may have given up on that route, but it was exactly for scenarios like this that the functionality was designed (that is, to leave the logic on the server and keep the check in Calypso as a basic can-or-can't question instead of a compound expression)

Reply:
I've updated the endpoint here: D27723-code. If you think the approach is sound, I'll add the necessary reducers/selectors to make it work.

Reply:
> We may have given up on that route, but it was exactly for scenarios like this that the functionality was designed (that is, to leave the logic on the server and keep the check in Calypso as a basic can-or-can't question instead of a compound expression)

Maybe @kwight can provide some more context here?